src/unread/__tests__/unreadModel-test.js

```diff
@@ -212,9 +220,38 @@ describe('stream substate', () => {
       expect(newState.streams.get(234)).toBe(state.streams.get(234));
     });
 
-    test('when operation is "remove" do nothing', () => {
-      const action = mkAction({ messages: [1, 2], op: 'remove' });
-      expect(reducer(baseState, action, eg.plusReduxState)).toBe(baseState);
+    test('when operation is "remove", add stream messages to unreads', () => {
+      const action = mkAction({
+        messages: [42, 99],
+        op: 'remove',
+        message_info_dict: new Map([
+          [
+            42,
+            {
+              type: 'stream',
+              mentioned: false,
+              sender_id: eg.selfUser.user_id,
+              stream_id: 123,
+              topic: 'foo',
+              unmuted_stream_msg: true,
+            },
+          ],
+          [
+            99,
+            {
+              type: 'stream',
+              mentioned: false,
+              sender_id: eg.selfUser.user_id,
+              stream_id: 123,
+              topic: 'foo',
+              unmuted_stream_msg: true,
+            },
+          ],
+        ]),
+      });
+      const newState = reducer(baseState, action, eg.plusReduxState);
+      expect(newState.streams.valueSeq().flatten()).toContain(42);
+      expect(newState.streams.valueSeq().flatten()).toContain(99);
```

Comment:
Let's make these expectations more specific. The `summary` helper used elsewhere in this file makes it pretty clean and compact to just say exactly what the state should look like.

One bit that would be good to include in a test case is that the IDs for a given conversation should be sorted, even if that means they're interleaved with the existing ones.